sofa-ark-parent/core-impl/container/src/main/java/com/alipay/sofa/ark/container/service/extension/ExtensionLoaderServiceImpl.java

```diff
@@ -0,0 +1,194 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.alipay.sofa.ark.container.service.extension;
+
+import com.alipay.sofa.ark.common.log.ArkLoggerFactory;
+import com.alipay.sofa.ark.common.util.OrderComparator;
+import com.alipay.sofa.ark.common.util.StringUtils;
+import com.alipay.sofa.ark.exception.ArkException;
+import com.alipay.sofa.ark.spi.model.Plugin;
+import com.alipay.sofa.ark.spi.service.extension.Extensible;
+import com.alipay.sofa.ark.spi.service.extension.Extension;
+import com.alipay.sofa.ark.spi.service.extension.ExtensionClass;
+import com.alipay.sofa.ark.spi.service.extension.ExtensionLoaderService;
+import com.alipay.sofa.ark.spi.service.plugin.PluginManagerService;
+import org.slf4j.Logger;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static com.alipay.sofa.ark.spi.constant.Constants.EXTENSION_FILE_DIR;
+
+/**
+ * @author qilong.zql
+ * @since 0.6.0
+ */
+@Singleton
+public class ExtensionLoaderServiceImpl implements ExtensionLoaderService {
+    private static final ConcurrentHashMap<Class, ConcurrentHashMap<String, ExtensionClass>> EXTENSION_MAP = new ConcurrentHashMap<>();
```

Comment:
Just a suggestion: should we store plugin name here? when use telnet to show plugin info ,can should extension being loaded as well.